File: On_First_Four_Bits/On.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 01/09/2023
    Program : Write a program which accept one number from user and on its first 4
              bits. Return modified number.

              Input : 73
              Output : 79 
*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
package main
import "fmt"

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To On First Four Bits 
func OnBit(iNo int) int {
	var iResult int = 0;
	var iMask int = 0X0000000F;

	if (iNo < 0) {
		iNo = -iNo;
	}

	iResult = iNo | iMask;

	return iResult;
}	

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
func main() {
	var iret int = 0;
	var iNo int = 0;

	fmt.Println("Jay Ganesh....");
	
	//Taking Input from user
	fmt.Println("Enter One Number");
	fmt.Scan(&iNo);

	//Call To OnBit Function
	iret = OnBit(iNo);

	//Printing Modified Number
	fmt.Println("Modified Number is :",iret);
}

File: On_First_Four_Bits/On.c
/*
    Author : Pakshal Shashikant Jain 
    Date : 01/09/2023
    Program : Write a program which accept one number from user and on its first 4
              bits. Return modified number.

              Input : 73
              Output : 79 
*/

//Required Libraries
#include<stdio.h>

typedef unsigned int UINT;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To On First Four Bits 
UINT OnBit(UINT iNo)
{
    UINT iResult = 0;
    int iMask = 0X0000000F;

    iResult = iNo | iMask;

    return iResult;
}

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

int main()
{
    UINT iret = 0;
    int iNo = 0;

    printf("Jay Ganesh....\n");

    //Taking Input from user
    printf("Enter One Number..\n");
    scanf("%d",&iNo);

    //Call To OnBit function
    iret = OnBit(iNo);

    //Printing Modifed Number
    printf("Modified Number is : %u",iret);

    return 0;
}